### ISSUE TITLE
Peptide AI/ML Assurance v0.1 — source verification, leakage-resistant splits, and closed-loop receipts

### ISSUE THREAD

**Issue by hummbl-dev** [hummbl-io/ai-source-verification#19]
Parent coordination: https://github.com/hummbl-dev/hummbl-dev/issues/145
Source registry: https://github.com/hummbl-dev/hummbl-bibliography/issues/77
Claim/evidence schema: https://github.com/hummbl-dev/claim-evidence-ledger/issues/8
Seed records: https://github.com/hummbl-dev/claim-evidence-ledger/issues/9
Ontology/API: https://github.com/hummbl-dev/knowledge-as-code/issues/6

## Status

**CANDIDATE ASSURANCE PROTOCOL — NO MODEL RELEASE — NON-CANONICAL**

## Objective

Create a reusable assurance protocol for AI/ML work on peptide data. The protocol must prevent literature claims, database annotations, assay labels, generated sequences, and benchmark scores from being treated as interchangeable evidence.

The first phase should define source verification, dataset lineage, split construction, benchmark reporting, model-output posture, and experiment-to-ledger receipts. It should not train or publish a peptide-generation model.

## Primary risks

Peptide datasets are especially vulnerable to:

- exact and near-duplicate sequences;
- variants from the same parent scaffold split across train/test;
- homologous peptide families spanning splits;
- duplicated records with conflicting labels;
- incompatible assays collapsed into one target variable;
- positive-only databases paired with synthetic negatives;
- missing concentration, matrix, pH, salt, serum, species, cell line, route, or time;
- publication and patent-selection bias;
- labels copied recursively between databases;
- uncertain modifications, topology, stereochemistry, or preparation identity;
- censored measurements and arbitrary thresholding;
- source leakage through embeddings, retrieved context, or foundation-model pretraining;
- test-set tuning and benchmark overfitting;
- generated candidates evaluated only by correlated predictors.

## Required dataset manifest

Define a versioned dataset card/manifest with at least:

```yaml
dataset_id:
version:
purpose:
task_definition:
source_records:
source_licenses:
entity_identity_level:
peptidoform_resolution:
assay_families:
label_definition:
units_and_normalization:
missingness_policy:
duplicate_policy:
conflict_policy:
negative_data_posture:
train_validation_test_split:
split_algorithm:
sequence_similarity_thresholds:
scaffold_or_family_groups:
time_cutoff:
source_holdouts:
known_leakage_risks:
exclusions:
preprocessing_code_ref:
checksums:
created_by:
reviewed_by:
as_of:
```

## Required split strategies

Implement or specify tests for:

- exact-sequence deduplication;
- normalized peptidoform deduplication;
- sequence-cluster split;
- scaffold/parent-series split;
- family or homology split;
- source/publication split;
- assay-laboratory split where metadata permits;
- temporal split;
- external-dataset holdout.

Random splitting may be included only as a diagnostic baseline and must not be presented as the primary generalization result without justification.

## Source verification rules

- Trace every label to an originating source, not only an aggregator database.
- Distinguish direct measurement from curator inference and computational prediction.
- Record whether multiple database entries ultimately cite the same experiment.
- Preserve assay conditions and units before harmonization.
- Do not manufacture negatives without labeling the negative-generation method.
- Treat absence from a database as unknown, not inactive.
- Record retractions, corrections, and superseded records.
- Hash and version data snapshots where licensing permits.

## Benchmark contract

Every benchmark report must include:

- task and decision relevance;
- dataset and split manifests;
- identity resolution level;
- baseline methods;
- metrics with confidence intervals or resampling uncertainty;
- calibration results where outputs are probabilistic;
- subgroup performance by sequence family, length, topology, modification state, assay family, and source when feasible;
- false-positive and false-negative analysis;
- out-of-distribution and abstention behavior;
- failure examples;
- leakage audit;
- model, code, environment, and seed receipts;
- claim language permitted by the evidence.

Do not allow one leaderboard metric to stand in for scientific utility.

## Multi-objective design posture

A peptide candidate may require simultaneous consideration of:

```text
target activity
selectivity
solubility
aggregation
protease stability
permeability or desired extracellular restriction
hemolysis/cytotoxicity
immunogenicity
PK/exposure
synthetic tractability
purification
formulation stability
cost
freedom to operate
```

Represent Pareto trade-offs and uncertainty. Do not collapse these into an opaque scalar without publishing the weighting and sensitivity analysis.

## Prediction / observation boundary

Model outputs must create `Prediction` records. Physical synthesis, identity confirmation, and assays create distinct `Observation` records. A prediction may be linked to later evidence but cannot be rewritten in place as an observation.

Required closed-loop lineage:

```text
hypothesis
→ candidate generation
→ synthesis-feasibility gate
→ physical production
→ identity/purity verification
→ assay with controls
→ atomic observations and claims
→ uncertainty-aware update
→ next-candidate decision
```

## Agent provenance

For any agent-generated dataset transformation, annotation, literature synthesis, or candidate recommendation, record:

- task specification;
- model/provider/version if available;
- retrieval sources and snapshots;
- tools and code versions;
- transformation steps;
- human/agent review roles;
- rejected outputs;
- downstream artifacts;
- permissions and private/public posture.

Agreement among multiple agents using shared sources is not independent replication.

## Adversarial fixtures

Include cases for:

- 95% sequence-identical peptides split across train/test;
- same scaffold with single-residue variants across splits;
- same nominal sequence but different modification or cyclization;
- positive and negative labels from incompatible salt/serum conditions;
- duplicated labels derived from the same paper through different databases;
- synthetic negatives mislabeled as experimentally inactive;
- retrospective threshold chosen after inspecting the test set;
- generated candidate scored by two models trained on overlapping data;
- prediction accidentally emitted as an observed claim;
- model benchmark with no source-license or split receipt.

## Acceptance criteria

- [ ] Dataset-card and split-manifest schemas are machine-valid.
- [ ] Leakage audit detects exact, near-duplicate, scaffold, and family overlap.
- [ ] Random split is visibly classified as a weak diagnostic when stronger splits are possible.
- [ ] Direct observations, curated labels, and predictions remain distinguishable.
- [ ] Assay incompatibility is detected or explicitly waived with rationale.
- [ ] Negative-data provenance is mandatory.
- [ ] Benchmark reports include calibration, uncertainty, subgroup, and failure analysis.
- [ ] Agent provenance resolves to source and transformation receipts.
- [ ] At least one valid and multiple adversarial fixtures are included.
- [ ] Output claim templates prevent benchmark inflation and therapeutic overreach.
- [ ] Independent reviewer performs a leakage/source-lineage red-team pass.
- [ ] PR records commands, outputs, environment, and reviewer disposition.

## Non-goals

- Training a production model.
- Generating actionable human-use peptide recommendations.
- Declaring benchmark superiority from random splits alone.
- Publishing restricted datasets.
- Treating model agreement as experimental confirmation.
- Creating a single universal developability score.

## Receipt requirement

Report dataset/split schema versions, leakage cases detected, fixtures, benchmark-template outputs, provenance coverage, unresolved licensing/source issues, commit SHA, and adversarial review findings.
